### PR TITLE
Update test infra architecture diagram

### PR DESCRIPTION
**Pull request by BobyMCbobs**
The test-infra diagram has been out of date for quite some time. Discussion about this has been made over here at #13063.
The changes have been based off of https://docs.google.com/presentation/d/1HOQ2df_AT-vIuz-JNaJol2oiGq84m50h9T49_5WgEaI/edit#slide=id.g4a3c5dc660_0_539

### CODE REVIEW

File: docs/architecture.svg [hunk 1]
```diff
@@ -1,282 +1,3 @@
...
...7.92 179.92 489.25 179.88" fill="none" stroke="#000000" stroke-miterlimit="10" pointer-events="stroke"/><path d="M 458.98 214.01 L 465.98 210.51 L 464.23 214.01 L 465.99 217.51 Z" fill="#000000" stroke="#000000" stroke-miterlimit="10" pointer-events="all"/><path d="M 494.5 179.87 L 487.51 183.39 L 489.25 179.88 L 487.49 176.39 Z" fill="#000000" stroke="#000000" stroke-miterlimit="10" pointer-events="all"/><path d="M 856.97 263.52 Q 857 201.58 833.58 201.58 Q 810.17 201.58 810.22 139.66" fill="none" stroke="#000000" stroke-miterlimit="10" pointer-events="stroke"/><path d="M 856.97 268.77 L 853.47 261.77 L 856.97 263.52 L 860.47 261.77 Z" fill="#000000" stroke="#000000" stroke-miterlimit="10" pointer-events="all"/><path d="M 810.22 134.41 L 813.72 141.42 L 810.22 139.66 L 806.72 141.41 Z" fill="#000000" stroke="#000000" stroke-miterlimit="10" pointer-events="all"/></g><switch><g requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"/><a transform="translate(0,-5)" xlink:href="https://www.diagrams.net/doc/faq/svg-export-text-problems" target="_blank"><text text-anchor="middle" font-size="10px" x="50%" y="100%">Viewer does not support full SVG 1.1</text></a></switch></svg>
```
Comment (spiffxp) [inline docs/architecture.svg:3]: Can you make this have line breaks for ease of diffing or review?

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm
I am fine if you merge as is and address comment in followup. Anything is better than what we have!  Thank you so much for doing this!

As discussed at k8s infra the diagram this referenced is now also about 2y out of date.

Suggested followup in linked issue.

**Comment by BobyMCbobs** [comment]
> /approve
> /lgtm
> I am fine if you merge as is and address comment in followup. Anything is better than what we have! Thank you so much for doing this!
> 
> As discussed at k8s infra the diagram this referenced is now also about 2y out of date.
> 
> Suggested followup in linked issue.

Thank you, @spiffxp !
Once I have a greater understanding of all of the k8s test-infra things including the separations (service, build, etc...) I'll add that too.

File: docs/architecture.svg (hunk 1 above)
Comment (BobyMCbobs) [reply docs/architecture.svg]: Updated in 9bffcca89d.
This appears to be something with draw.io's exporting

**Comment by BobyMCbobs** [comment]
In 0fa9337, I updated the DOT file to be the source of truth of the SVG instead of the SVG.
In the future I might have an idea for some sort of automation tie in for generating it and then rendering it.

File: docs/architecture.dot [hunk 2]
```diff
@@ -2,30 +2,35 @@
...
+    labelloc="t"
+    graph[compound=true]
 
 	# external services
-	legend_ext, GCS, PubSub, GitHub, BigQuery [style=filled, fillcolor=gray, shape=square]
+	legend_ext, GCS, PubSub, "GitHub\nthrough\nghproxy", BigQuery [style=filled, fillcolor=gray, shape=square]
 
 	# things that host websites
-	legend_site, Triage, Mungegithub, Deck, Testgrid, Gubernator [shape=house, fillcolor=lightblue, style=filled]
+	legend_site Spyglass, Deck, "Testgrid (closed)", Gubernator, Velodrome, Boskos, Greenhouse [shape=house, fillcolor=lightblue, style=filled]
```
Comment (MushuEE) [inline docs/architecture.dot:13]: Velodrome is now a thing of the past, as well as the Influx AFAIK @spiffxp, hopefully we can replace them with BigQuery + DataStudio soon, I will remember to update this if that lands

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Comment by BobyMCbobs** [comment]
> /approve
> /lgtm

@spiffxp, thank you for the merge and review!

**Pull request merged**